### PR TITLE
apt.py: fix environment on non-english locales

### PR DESCRIPTION
**Pull request by joneuhauser**
##### SUMMARY
Same issue as https://github.com/ansible/ansible/pull/76542, but in this case the error is not a false-negative "change", but a false positive (idempotence violated). This MR fixes this.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
apt

##### ADDITIONAL INFORMATION
See https://github.com/ansible/ansible/pull/76542

### CODE REVIEW

File: lib/ansible/modules/apt.py [hunk 1]
```diff
@@ -1139,6 +1139,7 @@ def main():
         DEBIAN_FRONTEND='noninteractive',
         DEBIAN_PRIORITY='critical',
         LANG=locale,
+        LANGUGE=locale,
```
Comment (jborean93) [inline lib/ansible/modules/apt.py]: Shoulnd't this be `LANGUAGE`. Also I would have assumed `LANG`  or `LC_*` affects this as well. Have you verified this actually fixed your issue?

**Comment by jborean93** [comment]
Also is it possible to add a test for such a scenario to ensure it doesn't regress again in the future.

File: lib/ansible/modules/apt.py (hunk 1 above)
Comment (joneuhauser) [reply lib/ansible/modules/apt.py]: Yes, sorry for the typo. I will try to add a test on the weekend. It does fix the issue locally, as does adding 
```
environment:
        LANGUAGE: en_US.utf8
```
to the play.

**Comment by s-hertel** [comment]
@joneuhauser I want to add a test for this, but I'm struggling to reproduce the issue - can you share an example? Since these env vars only apply to run_command and I can't trigger the bug installing a nonexisting package name containing non-ascii characters, we'd need to add a package in `setup_deb_repo` for the `apt` test to use https://github.com/ansible/ansible/tree/devel/test/integration/targets/setup_deb_repo/files/package_specs. Package names have restrictions

> [Package](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-package) names (both source and binary, see Package) must consist only of lower case letters (a-z), digits (0-9), plus (+) and minus (-) signs, and periods (.). They must be at least two characters long and must start with an alphanumeric character.

(from https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-package), so we can't actually add a package containing non-ascii characters in the name. I've tried add non-ascii characters in other fields (like the description and maintainer) but it doesn't seem to cause issues without this patch.

needs_info

**Pull request closed without merging**